include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
@gdr-at-ms I'm happy with that as a compromise. Personally, I started with the compiler version and then hit exactly the conundrum I described. So to avoid leaving behind an important set of internal users while they (necessarily) make use of `_HAS_STD_BYTE` in the short term, I'll make that change and add a comment pointing out that `_HAS_STD_BYTE` is likely to go away one day. I'll add in the GCC and Clang support now too, given I'm going to edit the %$#%# thing anyway.